src/System.Runtime/tests/System/StringTests.netcoreapp.cs

```diff
@@ -123,5 +124,156 @@ public static void Join_Char_InvalidStartIndexCount_ThrowsArgumentOutOfRangeExce
+                Assert.Equal("\u0069a", source.Replace("\u0130", "a", StringComparison.CurrentCultureIgnoreCase));
+            });
+        }
+
+        public static IEnumerable<object[]> Replace_StringComparisonCulture_TestData()
+        {
+            yield return new object[] { "abc", "abc", "def", false, null, "def" };
+            yield return new object[] { "abc", "ABC", "def", false, null, "abc" };
+            yield return new object[] { "abc", "abc", "def", false, CultureInfo.InvariantCulture, "def" };
+            yield return new object[] { "abc", "ABC", "def", false, CultureInfo.InvariantCulture, "abc" };
+
+            yield return new object[] { "abc", "abc", "def", true, null, "def" };
+            yield return new object[] { "abc", "ABC", "def", true, null, "def" };
+            yield return new object[] { "abc", "abc", "def", true, CultureInfo.InvariantCulture, "def" };
+            yield return new object[] { "abc", "ABC", "def", true, CultureInfo.InvariantCulture, "def" };
+
+            yield return new object[] { "abc", "abc" + SoftHyphen, "def", false, null, "def" };
+            yield return new object[] { "abc", "abc" + SoftHyphen, "def", true, null, "def" };
+            yield return new object[] { "abc", "abc" + SoftHyphen, "def", false, CultureInfo.InvariantCulture, "def" };
+            yield return new object[] { "abc", "abc" + SoftHyphen, "def", true, CultureInfo.InvariantCulture, "def" };
+
+            yield return new object[] { "\u0069\u0130", "\u0069", "a", false, new CultureInfo("tr-TR"), "a\u0130" };
+            yield return new object[] { "\u0069\u0130", "\u0069", "a", true, new CultureInfo("tr-TR"), "aa" };
+            yield return new object[] { "\u0069\u0130", "\u0069", "a", false, CultureInfo.InvariantCulture, "a\u0130" };
+            yield return new object[] { "\u0069\u0130", "\u0069", "a", true, CultureInfo.InvariantCulture, "a\u0130" };
+        }
+
+        [Theory]
+        [MemberData(nameof(Replace_StringComparisonCulture_TestData))]
+        public void Replace_StringComparisonCulture_ReturnsExpected(string original, string oldValue, string newValue, bool ignoreCase, CultureInfo culture, string expected)
+        {
+            Assert.Equal(expected, original.Replace(oldValue, newValue, ignoreCase, culture));
+            if (culture == null)
+            {
+                Assert.Equal(expected, original.Replace(oldValue, newValue, ignoreCase, CultureInfo.CurrentCulture));
+            }
+        }
+
+        [Fact]
+        public void Replace_StringComparsion_NullOldValue_ThrowsArgumentException()
+        {
+            Assert.Throws<ArgumentNullException>("oldValue", () => "abc".Replace(null, "def", StringComparison.CurrentCulture));
+            Assert.Throws<ArgumentNullException>("oldValue", () => "abc".Replace(null, "def", true, CultureInfo.CurrentCulture));
+        }
+
+        //TODO: [Fact]
```

Comment:
This currently fails with an OOM exception: https://github.com/dotnet/coreclr/pull/11001